src/App.tsx

```diff
@@ -15,48 +16,132 @@ export const goodsFromServer = [
   'Garlic',
 ];
 
+enum SortFieldValues {
+  default = '',
+  length = 'length',
+  abc = 'alphabetically',
+}
+
+interface ListObserverState {
+  sortType: SortFieldValues;
+  reverse: boolean;
+}
+
+function getPreparedGoods(
+  goods: string[],
+  sortType: SortFieldValues,
+  reverse: boolean,
+) {
+  let result = [...goods];
+
+  if (sortType) {
+    result.sort((good1, good2) => {
+      switch (true) {
+        case sortType === SortFieldValues.abc:
+          return good1.localeCompare(good2);
+
+        case sortType === SortFieldValues.length:
+          return good1.length - good2.length;
+
+        default:
+          return 0;
+      }
+    });
+  }
+
+  if (reverse) {
+    result = result.reverse();
+  }
+
+  return result;
+}
+
 export const App: React.FC = () => {
+  const [listObserver, setListObserver] = useState<ListObserverState>({
+    sortType: SortFieldValues.default,
+    reverse: false,
+  });
+
+  const preparedGoods = getPreparedGoods(
+    goodsFromServer,
+    listObserver.sortType,
+    listObserver.reverse,
+  );
+
+  const handleSort = (sortType: SortFieldValues) => {
```

Comment:
```suggestion
  const handleSort = (sortType: SortFieldValues) => () => {
```
handleSort can return arrow function. it is avoid to make anonmous arrow function in JSX